h2/connection.py

```diff
@@ -858,25 +858,35 @@ def send_data(self, stream_id, data, end_stream=False):
         :type data: ``bytes``
         :param end_stream: (optional) Whether this is the last data to be sent
             on the stream. Defaults to ``False``.
+        :param pad_length: (optional) Length of the padding to apply to the
+            data frame. Defaults to ``None`` for no use of padding. Note that
+            a value of ``0`` results in padding of length ``0``
+            (with the "padding" flag set on the frame).
         :type end_stream: ``bool``
         :returns: Nothing
         """
-        if len(data) > self.local_flow_control_window(stream_id):
+        frame_size = len(data)
+        if pad_length is not None:
+            frame_size += pad_length + 1
```

Comment:
Can we please validate the input? That is, we allow `pad_length` to be `None` or an integer that is `>= 0`. I'd like us to actually check that and to raise a `ValueError` if it's wrong.

Note that you can go ahead and raise the `ValueError` inside this `if` block if you want, which saves checking twice.